pkg/pub_package_reader/lib/pub_package_reader.dart

```diff
@@ -832,3 +833,20 @@ Iterable<ArchiveIssue> checkTopics(String pubspecContent) sync* {
     }
   }
 }
+
+Iterable<ArchiveIssue> checkHooks(Set<String> fileNames) sync* {
```

Comment:
Should this use the SDK version as @jonasfj suggested in the issue?

Reply:
Yeah, if we could pass along the SDK constraint and require that this greater than `3.6.0-0` ensuring that it's at-least Dart 3.6, then I think that'd help a bit.

Reply:
@dcharkes we don't know what SDK it'll land in yet, right? Just that it definitely won't be 3.5

Reply:
Updated, PTAL.